bigflow/build/spec.py

```diff
@@ -0,0 +1,262 @@
+"""Read and parse bigflow project configuration (setup.py / pyproject.toml)"""
+
+import functools
+import textwrap
+import typing
+import logging
+import typing
+import dataclasses
+import toml
+
+from pathlib import Path
+
+import setuptools
+
+import bigflow.resources
+import bigflow.version
+
+import bigflow.build.pip
+import bigflow.build.dev
+import bigflow.build.dataflow.dependency_checker
+
+import bigflow.commons as bf_commons
+
+
+logger = logging.getLogger(__name__)
+
+
+_ListStr = typing.List[str]
+
+# copied from `distutils.dist.DistributionMetadata`
+_PROJECT_METAINFO_KEYS = (
+    "name", "version", "author", "author_email",
+    "maintainer", "maintainer_email", "url",
+    "license", "description", "long_description",
+    "keywords", "platforms", "fullname", "contact",
+    "contact_email", "classifiers", "download_url",
+    # PEP 314
+    "provides", "requires", "obsoletes",
+)
+
+
+@dataclasses.dataclass
+class BigflowProjectSpec:
+    """Parameters of bigflow project."""
+
+    # transients: not part of 'pyproject.toml'
+    project_dir: Path
+
+    # Basic setuptools info - required or semi-required
+    name: str
+    version: str
+    packages: _ListStr
+    requries: _ListStr
+
+    # Setuptools-speicfic but _patched_ (not replaced) by bigflow
+    data_files: typing.List
+
+    # Bigflow-specific information and options
+    docker_repository: str
+    resources_dir: str
+    deployment_config_file: str
+    project_requirements_file: str
+
+    # Known package metainformation (author, url, description etc, see _PROJECT_METAINFO_KEYS)
+    metainfo: typing.Dict[str, typing.Any]
+
+    # Just bypass any unknow options to 'distutils'
```

Comment:
Just bypass any unknow**n** options to 'distutils'

Reply:
fixed